### PR TITLE
Automatic tests for updated edm4hep starterkit pages

### PR DESCRIPTION
**Pull request by vvolkl**
* change directory structure in preparation of moving those to fccanalyses / k4simdelphes
* minor fixes to code cells, to allow running the pages in the ci
* update the ci to run the pages in the working dir (to use `analysis.py` etc)
* Add code that follows the instruction "write to file xyz.py:" to an actual file. This is now duplicated - to avoid that there are  a few options (tbd in a future pr)
   - use the `includeliteral` directive from rst in the markdown
   - do not instruct to write to file, but add a line of python code at the end of the cell that acually runs it
   - use %%writefile magic from ipython (but this will give an error if copy and pasted to normal python)
 * use single backticks for inline code blocks
 * for the analysis page, use samples from eos (to make tests independent)
    - it would still be nice to have a check at the end of the delphes page to make sure that the files can be used with the analysis page. future pr ...

### CODE REVIEW

File: fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.md [hunk 1]
```diff
@@ -232,30 +239,31 @@ legend['VV'] = 'VV boson'
 ```
 
 And run the plots (the line below is temporary, will be updated once we properly link the env var)
```
Comment (clementhelsens) [inline fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.md:241]: could you also remove the comment in parenthesis?

**Review by clementhelsens — COMMENTED** [review]
just one comment to remove and good to merge

File: fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.md (hunk 1 above)
Comment (vvolkl) [reply fast-sim-and-analysis/fccanalyses/doc/starterkit/FccFastSimAnalysis/Readme.md]: good catch, thanks

**Review by clementhelsens — APPROVED** [review]

**Comment by clementhelsens** [comment]
If @gganis is ok it'd be good to merge now and say the new starter kits are ready at the coming meetings

**Comment by vvolkl** [comment]
@clementhelsens  Well, I'm still fixing herwig for the generators page, but that should be done soon. The ParticleMultiplicity can be done quickly, but the full simulation pages require a few fixes that will still take some time. I'm not really sure if it's not better to remove the driftchamber one alltogether, as the current simulations are done with a different implementaton

**Comment by vvolkl** [comment]
the nightlies are failing for the fast sim tutorial because they haven't updated since a breaking change was introduced in podio, I'm fixing those when the generators are working

**Comment by gganis** [comment]
I will approve this, no problem.

I have a comment, though. We are documenting the standalone application for Delphes, which is what we were using so far. However, we have now - in experimental form, I agree, but usable, I believe - the framework integration, which I believe should be our final goal.
Because we risk to end up having a plethora of applications, each with its way of passing arguments, which I think defeats the purpose of having a framework which unifies everything.

Ideally, since the Delphes framework integration is available when initialising key4hep, if not too complicated, I would add a section showing the same example with `k4run simdelphes.py --DelphesCard card_IDEA.tcl --DelphesOutputSettings edm4hep_IDEA.tcl --pythiacard p8_noBES_ee_ZH_ecm240.cmd --filename p8_ee_ZH_ecm240_edm4hep.root` . We need of course to provide `simdelphes.py`, which would be https://github.com/key4hep/k4SimDelphes/blob/main/framework/k4SimDelphes/examples/options/k4simdelphesalg.py with Pythia input instead of the particle gun.

**Comment by clementhelsens** [comment]
When I tried to read in a edm4hep file after k4Gen, it did not liked it. I did not tried within a same job to call k4gen instead of the particle gun, but if it works fine and the edm4hep outputs are consistent with the standalone application, we could even replace k4SimDelphes with the framework

**Comment by gganis** [comment]
I have tried and it is true that PythiaInterface behaves badly. But the problem seems to be in PythiaInterface not in the rest.
The standalone application uses the Pythia interface of Delphes, which is a different one, although the way errors are handled  after each event is generated seems very similar. The difference must be in the Pythia initialization.
I believe this is a urgent thing to fix, given the role PythiaInterface has for us ...

**Comment by vvolkl** [comment]
I am aware of at least one problem with the k4simdelphes framework interface that is just tricky to fix due to the way the converter works. It's connected to the MC-Reco-Association which explains why it works for the simple pgun example. I can't exclude any problems, but I am not aware of any issue with the PythiaInterface. It's more complicated for sure, as Michele added user hooks, but it should work, let me know if there is an issue.

In any case, it's good to do this step by step, I managed to fix the issues with the generators and the nightlies, I'll take  a look at switching this to the framework once that workds.

**Pull request merged**